### PR TITLE
roachtest: update version map and test fixtures for 21.2

### LINKED ISSUE #58237: roachtest: schemachange/mixed-versions failed
[(roachtest).schemachange/mixed-versions failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2538659&tab=buildLog) on [master@cd62a02833eeb02d14944668d09ee54806927a2b](https://github.com/cockroachdb/cockroach/commits/cd62a02833eeb02d14944668d09ee54806927a2b):

```
		  -- stack trace:
		  | main.(*cluster).RunE
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2292
		  | main.(*cluster).Run
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2212
		  | main.runSchemaChangeWorkloadStep.func1
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/mixed_version_schemachange.go:67
		  | main.(*versionUpgradeTest).run
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/versionupgrade.go:189
		  | main.runSchemaChangeMixedVersions
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/mixed_version_schemachange.go:144
		  | main.registerSchemaChangeMixedVersions.func1
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/mixed_version_schemachange.go:36
		  | main.(*testRunner).runTest.func2
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:760
		  | runtime.goexit
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1374
		Wraps: (2) output in run_071010.723_n3_workload_run_schemachange
		Wraps: (3) /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-2538659-1608706971-01-n4cpu4:3 -- ./workload run schemachange --verbose=1 --tolerate-errors=true --max-ops 100 --concurrency 5 {pgurl:1-4} returned
		  | stderr:
		  | <... some data truncated by circular buffer; go to artifacts for details ...>
		  | iled to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | W201223 08:09:45.161400 1 workload/cli/run.go:368  retrying after error while creating load: failed to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | W201223 08:09:47.114697 1 workload/cli/run.go:368  retrying after error while creating load: failed to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | W201223 08:09:49.365989 1 workload/cli/run.go:368  retrying after error while creating load: failed to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | W201223 08:09:51.304176 1 workload/cli/run.go:368  retrying after error while creating load: failed to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | W201223 08:09:53.377493 1 workload/cli/run.go:368  retrying after error while creating load: failed to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | W201223 08:09:55.339039 1 workload/cli/run.go:368  retrying after error while creating load: failed to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | W201223 08:09:57.176933 1 workload/cli/run.go:368  retrying after error while creating load: failed to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | W201223 08:09:59.451290 1 workload/cli/run.go:368  retrying after error while creating load: failed to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | W201223 08:10:01.628243 1 workload/cli/run.go:368  retrying after error while creating load: failed to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | W201223 08:10:03.869135 1 workload/cli/run.go:368  retrying after error while creating load: failed to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | W201223 08:10:05.956825 1 workload/cli/run.go:368  retrying after error while creating load: failed to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | W201223 08:10:08.175993 1 workload/cli/run.go:368  retrying after error while creating load: failed to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | W201223 08:10:10.238476 1 workload/cli/run.go:368  retrying after error while creating load: failed to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | W201223 08:10:12.009228 1 workload/cli/run.go:368  retrying after error while creating load: failed to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | E201223 08:10:12.148153 1 workload/cli/run.go:382  Attempt to create load generator failed. It's been more than 1h0m0s since we started trying to create the load generator so we're giving up. Last failure: failed to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | Error: failed to initialize the load generator: ERROR: desc 291: type ID 177 in descriptor not found: descriptor not found (SQLSTATE XXUUU)
		  | Error: COMMAND_PROBLEM: exit status 1
		  | (1) COMMAND_PROBLEM
		  | Wraps: (2) Node 3. Command with error:
		  |   | ```
		  |   | ./workload run schemachange --verbose=1 --tolerate-errors=true --max-ops 100 --concurrency 5 {pgurl:1-4}
		  |   | ```
		  | Wraps: (3) exit status 1
		  | Error types: (1) errors.Cmd (2) *hintdetail.withDetail (3) *exec.ExitError
		  |
		  | stdout:
		Wraps: (4) exit status 20
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *main.withCommandDetails (4) *exec.ExitError

```

<details><summary>More</summary><p>

Artifacts: [/schemachange/mixed-versions](https://teamcity.cockroachdb.com/viewLog.html?buildId=2538659&tab=artifacts#/schemachange/mixed-versions)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Aschemachange%2Fmixed-versions.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

### LINKED ISSUE #64525: roachtest: schemachange/database-version-upgrade failed
roachtest.schemachange/database-version-upgrade [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2944550&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2944550&tab=artifacts#/schemachange/database-version-upgrade) on master @ [684e753c15f3fc58df79b6ea70e7b6715eae4835](https://github.com/cockroachdb/cockroach/commits/684e753c15f3fc58df79b6ea70e7b6715eae4835):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/schemachange/database-version-upgrade/run_1
	schema_change_database_version_upgrade.go:51,schema_change_database_version_upgrade.go:38,test_runner.go:777: prev version not set for version: 21.2
		(1) attached stack trace
		  -- stack trace:
		  | main.PredecessorVersion
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:1210
		  | main.runSchemaChangeDatabaseVersionUpgrade
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/schema_change_database_version_upgrade.go:49
		  | main.registerSchemaChangeDatabaseVersionUpgrade.func1
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/schema_change_database_version_upgrade.go:38
		  | main.(*testRunner).runTest.func2
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:777
		  | runtime.goexit
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1374
		Wraps: (2) prev version not set for version: 21.2
		Error types: (1) *withstack.withStack (2) *errutil.leafError
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh schemachange/database-version-upgrade
```
</p>
</p>
</details>
/cc @cockroachdb/sql-schema
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*schemachange/database-version-upgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

### LINKED ISSUE #64537: roachtest: schemachange/secondary-index-multi-version failed
roachtest.schemachange/secondary-index-multi-version [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2944550&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2944550&tab=artifacts#/schemachange/secondary-index-multi-version) on master @ [684e753c15f3fc58df79b6ea70e7b6715eae4835](https://github.com/cockroachdb/cockroach/commits/684e753c15f3fc58df79b6ea70e7b6715eae4835):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/schemachange/secondary-index-multi-version/run_1
	secondary_indexes.go:141,test_runner.go:777: prev version not set for version: 21.2
		(1) attached stack trace
		  -- stack trace:
		  | main.PredecessorVersion
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:1210
		  | main.registerSecondaryIndexesMultiVersionCluster.func1
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/secondary_indexes.go:139
		  | main.(*testRunner).runTest.func2
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:777
		  | runtime.goexit
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1374
		Wraps: (2) prev version not set for version: 21.2
		Error types: (1) *withstack.withStack (2) *errutil.leafError
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh schemachange/secondary-index-multi-version
```
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #63954 roachtest: schemachange/secondary-index-multi-version failed [C-test-failure O-roachtest O-robot branch-release-20.2 release-blocker]
</p>
</details>
/cc @cockroachdb/sql-schema
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*schemachange/secondary-index-multi-version.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

### LINKED ISSUE #64594: roachtest: decommission/mixed-versions failed
roachtest.decommission/mixed-versions [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2945603&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2945603&tab=artifacts#/decommission/mixed-versions) on master @ [dbf711a83969210bcf706d65f53f888856da6ad0](https://github.com/cockroachdb/cockroach/commits/dbf711a83969210bcf706d65f53f888856da6ad0):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/decommission/mixed-versions/run_1
	mixed_version_decommission.go:31,decommission.go:73,test_runner.go:777: prev version not set for version: 21.2
		(1) attached stack trace
		  -- stack trace:
		  | main.PredecessorVersion
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:1210
		  | main.runDecommissionMixedVersions
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/mixed_version_decommission.go:29
		  | main.registerDecommission.func3
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/decommission.go:73
		  | main.(*testRunner).runTest.func2
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:777
		  | runtime.goexit
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1374
		Wraps: (2) prev version not set for version: 21.2
		Error types: (1) *withstack.withStack (2) *errutil.leafError
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh decommission/mixed-versions
```
</p>
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #58058 roachtest: decommission/mixed-versions failed [C-test-failure O-roachtest O-robot branch-release-20.2]
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*decommission/mixed-versions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

### LINKED ISSUE #64595: roachtest: import/mixed-versions failed
roachtest.import/mixed-versions [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2945603&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2945603&tab=artifacts#/import/mixed-versions) on master @ [dbf711a83969210bcf706d65f53f888856da6ad0](https://github.com/cockroachdb/cockroach/commits/dbf711a83969210bcf706d65f53f888856da6ad0):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/import/mixed-versions/run_1
	import.go:298,test_runner.go:777: prev version not set for version: 21.2
		(1) attached stack trace
		  -- stack trace:
		  | main.PredecessorVersion
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:1210
		  | main.registerImportMixedVersion.func1
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/import.go:296
		  | main.(*testRunner).runTest.func2
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:777
		  | runtime.goexit
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1374
		Wraps: (2) prev version not set for version: 21.2
		Error types: (1) *withstack.withStack (2) *errutil.leafError
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh import/mixed-versions
```
</p>
</p>
</details>
/cc @cockroachdb/bulk-io
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*import/mixed-versions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

### LINKED ISSUE #64596: roachtest: autoupgrade failed
roachtest.autoupgrade [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2945603&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2945603&tab=artifacts#/autoupgrade) on master @ [dbf711a83969210bcf706d65f53f888856da6ad0](https://github.com/cockroachdb/cockroach/commits/dbf711a83969210bcf706d65f53f888856da6ad0):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/autoupgrade/run_1
	autoupgrade.go:252,test_runner.go:777: prev version not set for version: 21.2
		(1) attached stack trace
		  -- stack trace:
		  | main.PredecessorVersion
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:1210
		  | main.registerAutoUpgrade.func2
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/autoupgrade.go:250
		  | main.(*testRunner).runTest.func2
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:777
		  | runtime.goexit
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1374
		Wraps: (2) prev version not set for version: 21.2
		Error types: (1) *withstack.withStack (2) *errutil.leafError
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh autoupgrade
```
</p>
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*autoupgrade.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

### LINKED ISSUE #64603: roachtest: version/mixed/nodes=3 failed
roachtest.version/mixed/nodes=3 [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2945603&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2945603&tab=artifacts#/version/mixed/nodes=3) on master @ [dbf711a83969210bcf706d65f53f888856da6ad0](https://github.com/cockroachdb/cockroach/commits/dbf711a83969210bcf706d65f53f888856da6ad0):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/version/mixed/nodes=3/run_1
	version.go:220,test_runner.go:777: prev version not set for version: 21.2
		(1) attached stack trace
		  -- stack trace:
		  | main.PredecessorVersion
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:1210
		  | main.registerVersion.func2
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/version.go:218
		  | main.(*testRunner).runTest.func2
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:777
		  | runtime.goexit
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1374
		Wraps: (2) prev version not set for version: 21.2
		Error types: (1) *withstack.withStack (2) *errutil.leafError
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh version/mixed/nodes=3
```
</p>
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*version/mixed/nodes=3.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

### LINKED ISSUE #64611: roachtest: version/mixed/nodes=5 failed
roachtest.version/mixed/nodes=5 [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2945603&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2945603&tab=artifacts#/version/mixed/nodes=5) on master @ [dbf711a83969210bcf706d65f53f888856da6ad0](https://github.com/cockroachdb/cockroach/commits/dbf711a83969210bcf706d65f53f888856da6ad0):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/version/mixed/nodes=5/run_1
	version.go:220,test_runner.go:777: prev version not set for version: 21.2
		(1) attached stack trace
		  -- stack trace:
		  | main.PredecessorVersion
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:1210
		  | main.registerVersion.func2
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/version.go:218
		  | main.(*testRunner).runTest.func2
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:777
		  | runtime.goexit
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1374
		Wraps: (2) prev version not set for version: 21.2
		Error types: (1) *withstack.withStack (2) *errutil.leafError
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh version/mixed/nodes=5
```
</p>
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*version/mixed/nodes=5.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

### LINKED ISSUE #64618: roachtest: tpcc/mixed-headroom/n5cpu16 failed
roachtest.tpcc/mixed-headroom/n5cpu16 [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2945603&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2945603&tab=artifacts#/tpcc/mixed-headroom/n5cpu16) on master @ [dbf711a83969210bcf706d65f53f888856da6ad0](https://github.com/cockroachdb/cockroach/commits/dbf711a83969210bcf706d65f53f888856da6ad0):

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/tpcc/mixed-headroom/n5cpu16/run_1
	tpcc.go:288,test_runner.go:777: prev version not set for version: 21.2
		(1) attached stack trace
		  -- stack trace:
		  | main.PredecessorVersion
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:1210
		  | main.registerTPCC.func2
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tpcc.go:286
		  | main.(*testRunner).runTest.func2
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:777
		  | runtime.goexit
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1374
		Wraps: (2) prev version not set for version: 21.2
		Error types: (1) *withstack.withStack (2) *errutil.leafError
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh tpcc/mixed-headroom/n5cpu16
```
</p>
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*tpcc/mixed-headroom/n5cpu16.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

### PR DESCRIPTION
**Pull request by j-low**
Fixes #64595.
Fixes #64594.
Fixes #64537.
Fixes #64525. 
Fixes #64596.
Fixes #58237.
Fixes #64603.
Fixes #64611.
Fixes #64210.
Fixes #64618.

Release note: None

### CODE REVIEW

File: pkg/cmd/roachtest/acceptance.go [hunk 1]
```diff
@@ -58,7 +58,6 @@ func registerAcceptance(r *testRegistry) {
 			// to head after 19.2 fails.
 			minVersion: "v19.2.0",
 			timeout:    30 * time.Minute,
-			skip:       "https://github.com/cockroachdb/cockroach/issues/64491",
```
Comment (j-low) [inline pkg/cmd/roachtest/acceptance.go:61]: Not sure if this line should be reverted?

Comment (irfansharif) [reply pkg/cmd/roachtest/acceptance.go:61]: Yup, it should.

**Review by irfansharif — APPROVED** [review]
Thanks! Feel free to close out all the linked issues once merged (or link to them in the commit message here to have it done for you).

**Comment by j-low** [comment]
@irfansharif TFTR! Will revert line before merge and close issues.

**Comment by j-low** [comment]
Fixes #64595 #64594 #64537 #64525 #64596 #58237 #64603

**Comment by irfansharif** [comment]
I see the fixtures are `checkpoint-v20.2-48.tgz`, I think we want them to be generated using the 21.1 beta build (it's currently tripping up CI).

**Comment by tbg** [comment]
v21.1.0-beta.5 unfortunately self-identifies as 20.2-48, i.e. it doesn't have the 20.2.0 cluster version yet. Unfortunately there also isn't a newer released version that has the correct cluster version. So we're looking at hacking it, I'll give that a shot.

**Comment by tbg** [comment]
I updated this PR by renaming the 20.2-48 fixture to 21.1. This should work until we bump the MinSupportedVersion, which we should only do when 21.1.0 is out the door.

**Comment by j-low** [comment]
TY @tbg and @irfansharif for helping get this across the line.

**Pull request merged**